operator/src/main/java/oracle/kubernetes/weblogic/domain/model/ClusterSpec.java

```diff
@@ -3,67 +3,293 @@
 
 package oracle.kubernetes.weblogic.domain.model;
 
-import java.util.Collections;
-import java.util.List;
 import java.util.Map;
 import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
 
-import io.kubernetes.client.openapi.models.V1Container;
-import io.kubernetes.client.openapi.models.V1PodSpec;
+import com.google.gson.annotations.Expose;
+import com.google.gson.annotations.SerializedName;
 import io.kubernetes.client.openapi.models.V1ServiceSpec;
+import oracle.kubernetes.json.Description;
+import oracle.kubernetes.json.EnumClass;
+import oracle.kubernetes.json.Range;
+import oracle.kubernetes.operator.ServerStartPolicy;
+import oracle.kubernetes.operator.ServerStartState;
+import org.apache.commons.lang3.builder.EqualsBuilder;
+import org.apache.commons.lang3.builder.HashCodeBuilder;
+import org.apache.commons.lang3.builder.ToStringBuilder;
 
-public abstract class ClusterSpec {
-
-  /**
-   * Returns the labels applied to the cluster service.
-   *
-   * @return a map of labels
-   */
+/**
+ * An element representing a cluster in the domain configuration.
+ *
+ * @since 2.0
+ */
+public class ClusterSpec extends BaseConfiguration implements Comparable<ClusterSpec> {
```

Comment:
I was expecting to see a field that referenced the Domain that this Cluster belongs to... or, is that only specified on the Domain side (i.e. Domain lists Cluster resources associated with the Domain)?

Reply:
At the moment, the associated Domain is referenced as a label in the Cluster Resource:

e.g.
apiVersion: "weblogic.oracle/v1"
kind: Cluster
metadata:
  name: cluster-1
  namespace: sample-domain1-ns
  labels:
    weblogic.domainUID: sample-domain1

I didn't add or modify the attributes of the Cluster spec from the original.  I've been identifying the associated domain by using the label.